### PR TITLE
feat: implement keynote page

### PR DESCRIPTION
**Pull request by mattwang44**
## Types of Changes

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description

Add keynote page (`<host>/<base>/<lang>/conference/keynotes`).

## More Information

https://user-images.githubusercontent.com/24987826/123493898-95917e00-d650-11eb-8949-cecc88ac1d42.mp4

### CODE REVIEW

File: components/core/tabs/Tabs.vue [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+</script>
+
+<style lang="postcss" scoped>
+.tabs__headers {
+    @apply ml-4;
+}
+.tabs__headers > .header {
+    @apply inline-block ml-1 py-0.5 md:py-2 px-1 md:px-4;
+    @apply text-xs md:text-sm text-center font-bold;
+    @apply border-2 rounded-t-lg bg-transparent cursor-pointer;
```
Comment (flynnhou) [inline components/core/tabs/Tabs.vue:53]:
```suggestion
    @apply border rounded-t-lg bg-transparent cursor-pointer;
```

to align with the design

File: components/core/tabs/Tabs.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        min-width: 110px;
+    }
+}
+.tabs__headers > .header.-selected {
+    background: #e6ba17;
+    color: #121023;
+}
+
+.tab {
+    @apply inline-block text-sm rounded-2xl min-w-full;
```
Comment (flynnhou) [inline components/core/tabs/Tabs.vue:68]:
```suggestion
    @apply inline-block text-sm rounded-2xl min-w-full border-2;
```

to align with the design.

File: components/core/tabs/Tabs.vue [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    color: #282828;
+    background: #e6ba17;
+    border-color: #c2a53a;
+    padding: 20px;
```
Comment (flynnhou) [inline components/core/tabs/Tabs.vue:72]: add the following css rule to align with the design:

```css
    box-shadow: 6px 6px 0 #c2a53a;
```

copied from @dabingnai's event card styling - https://github.com/dabingnai/pycontw-2021/blob/55d21044107cd6af9925f536ee1150874ba91b53/components/events/EventCard.vue#L41

**Review by flynnhou — CHANGES_REQUESTED** [review]

File: components/core/tabs/Tabs.vue [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+<script>
+export default {
+    data() {
+        return {
+            selectedIndex: 0,
+            tabs: [],
+        }
+    },
+    computed: {},
```
Comment (flynnhou) [inline components/core/tabs/Tabs.vue:25]: remove unused line

File: components/core/tabs/Tabs.vue [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    mounted() {
+        this.selectTab(0)
+    },
+    methods: {
+        selectTab(i) {
+            this.selectedIndex = i
+            this.tabs.forEach((tab, index) => {
+                tab.isActive = index === i
+            })
+        },
```
Comment (flynnhou) [inline components/core/tabs/Tabs.vue:38]: Love the implementation! Elegant!

File: pages/conference/keynotes.vue [hunk 6]
```diff
@@ -1,11 +1,179 @@
...
+        locale() {
+            return { 'en-us': 'en_us', 'zh-hant': 'zh_hant' }[this.$i18n.locale]
+        },
+    },
+    async mounted() {
+        await this.$store.dispatch('$getKeynotesData')
+        this.keynotesData = this.$store.state.keynotesData.map((keynote) => ({
+            ...keynote,
+            id: this.$makeId(),
+        }))
```
[the remark this answers is not in the document]
Comment (mattwang44) [reply pages/conference/keynotes.vue:130]: Directly use `$makeId()` in the loop results in error.

image

**Comment by mattwang44** [comment]
@flynnhou Thanks for reviewing!
Added a new commit e27b09d for `head()`.

**Review by flynnhou — APPROVED** [review]
lgtm

**Pull request merged**